File: LeetCode/Tag/Array/cpp/1539.kth-missing-positive-number.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        for (int i = 0, x = 1; i < arr.size(); ++x) {
            if (arr[i] == x)
                ++i;
            else if (--k == 0)
                return x;
        }
        return arr.back() + k;
    }
};

File: LeetCode/Difficulty/Easy/js/1539.Kth-missing-positive-number.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findKthPositive = function(arr, k) {
    let missingArr = []
        ,  i = 1
        ,  len = arr.length;
    let max = arr[len-1];

    missingArr = new Array(max).fill(0);
    arr.forEach(item=>missingArr[item]="test");

    missingArr.forEach((item,index)=>{
        if(item!=="test")
            missingArr[index] = index;
    });

    missingArr=missingArr.filter(item=>item!="test");

    if(missingArr.length<=1){
        return max+k;
    }
    else if(missingArr.length <= k){
        return max+Math.abs(k-(missingArr.length - 1));
    }

    return missingArr[k];
};